### PR TITLE
chore: Update package dependencies and add Copilot instructions

### PR DESCRIPTION
**Pull request by vmelamed**
# Summary

## Commits

- chore: Update package dependencies and add Copilot instructions

## Checklist

- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] Tests pass locally, and cover new code and edge cases. Test coverage is above 80%.
- [x] It is not expected that the performance will degrade by more than 20%
- [x] Documentation is updated if necessary

### CODE REVIEW

File: .github/copilot-instructions.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+## Shared Conventions
+
+Refer to [CONVENTIONS.md](CONVENTIONS.md) for the full set of coding, testing,
+documentation, and CI conventions shared across all vm2 packages.
+
+## Package-Specific Guidance
+
+_TODO_ Describe what this package does, key design decisions, invariants to preserve,
+and any Copilot-specific hints for working in this codebase.
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply .github/copilot-instructions.md:11]: ok
